InteractiveHtmlBom/web/render.js

```diff
@@ -483,21 +484,53 @@ function handlePointerLeave(e, layerdict) {
   delete layerdict.pointerStates[e.pointerId];
 }
 
+function resetTransform(layerdict) {
+  layerdict.transform.panx = 0;
+  layerdict.transform.pany = 0;
+  layerdict.transform.zoom = 1;
+  redrawCanvas(layerdict);
+}
+
 function handlePointerUp(e, layerdict) {
+  if (!e.hasOwnProperty("offsetX")) {
+    // The polyfill doesn't set this properly
+    e.offsetX = e.pageX - e.currentTarget.offsetLeft;
+    e.offsetY = e.pageY - e.currentTarget.offsetTop;
+  }
+
   e.preventDefault();
   e.stopPropagation();
 
-  if (e.button == 0 && layerdict.pointerStates.hasOwnProperty(e.pointerId) && layerdict.pointerStates[e.pointerId].distanceTravelled < 10) {
-    // This is just a click
-    handleMouseClick(e, layerdict);
+  if (layerdict.pointerStates.hasOwnProperty(e.pointerId)) {
+    // We haven't necessarily had a pointermove event since the interaction started, so make sure we update this now
```

Comment:
But we would always have PointerDown event, right? JS would not reorder them afaik.
So we can get rid of the `layerdict.pointerStates.hasOwnProperty(e.pointerId)` check or just add fast return to be safe. 
That would allow us to avoid repeating that check below and also reuse `ptr`.

Reply:
Good point. The worst that can happen is that the event handler throws an exception, which ends up being equivalent to an early return anyway :)